Framework/Kernel/src/EnabledWhenProperty.cpp

```diff
@@ -37,10 +37,9 @@ EnabledWhenProperty::EnabledWhenProperty(
     : // This method allows the Python interface to easily construct these
       // objects
       // Copy the object then forward onto our move constructor
-      EnabledWhenProperty(
-          std::move(std::make_shared<EnabledWhenProperty>(conditionOne)),
-          std::move(std::make_shared<EnabledWhenProperty>(conditionTwo)),
-          logicOperator) {}
+      EnabledWhenProperty(std::make_shared<EnabledWhenProperty>(conditionOne),
+                          std::make_shared<EnabledWhenProperty>(conditionTwo),
```

Comment:
In this case because we aren't using `MatrixWorkspace_sptr` and associated typedefs I opted to use `std::shared_ptr` so we don't need to refactor it at a later date if we use the boost version. 

`boost::python` handles this correctly and the previous PR added some unit tests to check the properties behave correctly just in case